### PR TITLE
Switch from fake storage to mock clients in azure unit tests.

### PR DESCRIPTION
**Pull request by nilo19**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:

Switch from fake storage to mock clients in azure unit tests.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/area provider/azure
/priority important-soon

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer_test.go [hunk 1]
```diff
@@ -21,20 +21,26 @@ package azure
...
 	"context"
 	"fmt"
+	"net/http"
 	"reflect"
 	"strconv"
 	"strings"
 	"testing"
 	"time"
 
+	"k8s.io/legacy-cloud-providers/azure/clients/loadbalancerclient/mockloadbalancerclient"
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer_test.go:31]: move below together with other k8s.io packages

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer_test.go [hunk 2]
```diff
@@ -900,7 +928,7 @@ func TestIsFrontendIPChanged(t *testing.T) {
 			lbFrontendIPConfigName: "btest1-name",
 			service:                getTestService("test1", v1.ProtocolTCP, nil, false, 80),
 			loadBalancerIP:         "1.1.1.1",
-			exsistingPIPs: []network.PublicIPAddress{
+			existingPIPs: []network.PublicIPAddress{
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer_test.go:931]: thanks for fixing typos

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/interface.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fileclient
+
+// Interface is the client interface for creating file shares, interface for test injection.
+// mockgen -source=$GOPATH/src/k8s.io/kubernetes/staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/interface.go -package=mockfileclient Interface > $GOPATH/src/k8s.io/kubernetes/staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/mockfileclient/interface.go
+type Interface interface {
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/interface.go:23]: @andyzhangx The file client is moved to a new package with this PR

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Comment by feiskyer** [comment]
@liggitt could you help to have a look at vendor changes?

**Comment by liggitt** [comment]
/approve
for vendor change

**Comment by saschagrunert** [comment]
Hey @nilo19, may I ask you to remove the period from the release notes block that we do not catch it up in the release notes parsing process? Like just:
```
None
```
😇

**Comment by nilo19** [comment]
> Hey @nilo19, may I ask you to remove the period from the release notes block that we do not catch it up in the release notes parsing process? Like just:
> 
> ```
> None
> ```
> 
> 😇

Removed, thank you.

**Pull request merged**